numpy/linalg/linalg.py

```diff
@@ -1142,6 +1144,8 @@ def eig(a):
     _assertNdSquareness(a)
```

Comment:
Lapack actually seems able to handle 0x0 (not 0d) in a lot of cases, but needs more careful argument preparation, that we are not doing. In particular, I think I can make `lstsq` work with it. I think your pr removing that weird error would become useful for the vectorized ops though.